src/Service.Tests/SqlTests/RestApiTests/Insert/MsSqlInsertApiTests.cs

```diff
@@ -261,6 +273,67 @@ public async Task TestCleanup()
 
         #endregion
 
+        /// <summary>
+        /// Test to validate successful execution of a request when a timestamp field is missing from the request body.
+        /// In such a case, we skip inserting the field. 
+        /// </summary>
+        [TestMethod]
+        public async Task InsertOneWithTimestampFieldMissingFromRequestBody()
+        {
+            // Validate successful execution of a POST request when a timestamp field (here 'row_version')
+            // is missing from the request body.
+            string requestBody = @"
+            {
+                ""id"": 2,
+                ""book_name"": ""Another Awesome Book"",
+                ""copies_sold"": 100,
+                ""last_sold_on"": null
+            }";
+            string expectedLocationHeader = $"id/2";
+
+            await SetupAndRunRestApiTest(
+                    primaryKeyRoute: null,
+                    queryString: null,
+                    entityNameOrPath: _entityWithReadOnlyFields,
+                    sqlQuery: GetQuery("InsertOneWithTimestampFieldMissingFromRequestBody"),
+                    operationType: EntityActionOperation.Insert,
+                    requestBody: requestBody,
+                    expectedStatusCode: HttpStatusCode.Created,
+                    expectedLocationHeader: expectedLocationHeader
+                );
+        }
+
+        /// <summary>
+        /// Test to validate that whenever a timestamp field is included in the request body, we throw a BadRequest exception
+        /// as it is not allowed to provide value (to insert) for a timestamp field.
+        /// </summary>
+        [TestMethod]
+        public async Task InsertOneWithTimestampFieldInRequestBody()
+        {
+            // Validate that a BadRequest exception is thrown when a timestamp field (here 'row_version') is included in request body.
+            string requestBody = @"
+            {
+                ""id"": 2,
+                ""book_name"": ""Another Awesome Book"",
+                ""copies_sold"": 100,
+                ""last_sold_on"": null,
+                ""row_version"": null
```

Comment:
Can we test assigning a non-null valid value?

Reply:
The example you provide in the .sql file for timestamp https://learn.microsoft.com/en-us/sql/relational-databases/xml/for-xml-support-for-the-timestamp-data-type?view=sql-server-ver16 shows an insert using value null for the timestamp field. so why it would not be okay here.. i don't understand.

Reply:
Added test for non-null value. For Sean's comment: https://github.com/Azure/data-api-builder/pull/1596#discussion_r1300331376.